lib/src/Uncertainty/Algorithm/Analytical/Analytical.cxx

```diff
@@ -43,7 +43,10 @@ Analytical::Analytical(const OptimizationAlgorithm & nearestPointAlgorithm,
     physicalStartingPoint_(physicalStartingPoint)
 {
   const UnsignedInteger dimension = event.getImplementation()->getFunction().getInputDimension();
-  if (physicalStartingPoint.getDimension() != dimension) throw InvalidArgumentException(HERE) << "Starting point dimension (" << physicalStartingPoint.getDimension() << ") does not match event dimension (" << dimension << ").";
+  if (physicalStartingPoint.getDimension() != dimension)
+    throw InvalidArgumentException(HERE) << "Starting point dimension (" << physicalStartingPoint.getDimension() << ") does not match event dimension (" << dimension << ").";
+  if (!event_.getImplementation()->getAntecedent().getDistribution().isContinuous())
```

Comment:
Be careful, discrete is not the opposite of being continuous. FORM/SORM methods are dedicated to absolutely continuous distributions, if only one marginal distribution is not continuous it will fail. Discrete is when **all** the marginals are discrete.

Reply:
yes, isContinuous will fail if at least one marginal is discrete

Reply:
exactly